scripts/manage_case

```diff
@@ -1,375 +1,172 @@
+    infile = files.get_value("CONFIG_DRV_FILE")
+    comps_object = Component(infile)
+    comps_array = comps_object.get_value("component")
+    
+    # Loop through the elements for each component class (in config_files.xml) 
+    # and see if there is a match for the the target component in the component attribute 
+    match_found = False
+    valid_components = list()
+    for comp in comps_array:
+        string = "CONFIG_%s_FILE"%comp 
+
+        # determine all components in string
+        components = files.get_components(string, resolved=True)
+        for item in components:
+            valid_components.append(item)
+
+        # determine if config_file is on disk
+        config_file = files.get_value(string, attribute={"component":name}, resolved=True)
+        if config_file is not None:
+            match_found = True
+            expect(os.path.isfile(config_file),
+                   "Cannot find config_file %s on disk" %config_file)
+            break
+
+    # If name is not a valid argument - exit with error
+    expect((match_found), 
+           "Invalid input argument %s, valid input arguments are %s" % (name, valid_components))
+
+    # Check that file exists on disk, if not exit with error
+    expect((config_file),
+           "Cannot find any config_component.xml file for %s" %name)
+
+    # determine component xml content 
+    parsed_xml = Component(config_file)
+    parsed_xml.print_values()
+
+###############################################################################
+def parse_command_line(args):
+###############################################################################
+
+    cime_model = CIME.utils.get_model()
+
+    parser = argparse.ArgumentParser()
+
+    CIME.utils.setup_standard_logging_options(parser)
+
+    parser.add_argument("--query_compsets_setby", 
+                        help="Query compsets that are set by the target component for %s model"%cime_model)
+
+    parser.add_argument("--query_component_name", 
+                        help="Query component settings that are set by the target component for %s model"%cime_model)
+
+    parser.add_argument("--query_grids", action="store_true",
+                        help="Query supported model grids for %s model" %cime_model)  
+
+    parser.add_argument("--grids_shortname", 
+                        help="Query model grids for input grid short name for %s model - not implemented yet" %cime_model)  
+
+    parser.add_argument("--grids_alias", 
+                        help="Query model grids for input grid  alias for %s model - not implemented yet" %cime_model)  
+
+    parser.add_argument("--query_machines", action="store_true", 
+                        help="Query supported machines for %s model" %cime_model)  
+
+    args = parser.parse_args()
+
+    CIME.utils.handle_standard_logging_options(args)
+
+    return args
+
+###############################################################################
+
+args = parse_command_line(sys.argv)
```

Comment:
It is generally not a good idea to do work in the global namespace. Please follow the convention we've used in our other scripts:
`def _main_func():
    *do stuff*

if __name__ == "__main__":
    _main_func()`